File: back/src/main/java/solutiona/challenge/pickaboo/domain/entity/DiseaseConst.java
package solutiona.challenge.pickaboo.domain.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.util.List;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@DynamicUpdate
@Table(name = "disease_const")
public class DiseaseConst {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "rating")
    private Long rating;

    @Column(name = "e_rating")
    private Long eRating;

    @Column(name = "name")
    private String name;

    @Column(name = "e_name")
    private String eName;

    @Column(name = "description")
    private String description;

    @Column(name = "e_description")
    private String eDescription;

    @Column(name = "type")
    private String type;

    @Column(name = "e_type")
    private String eType;

    @Column(name = "site")
    private String site;

    @Column(name = "e_site")
    private String eSite;

    @Column(name = "reason")
    private String reason;

    @Column(name = "e_reason")
    private String eReason;

    @Column(name = "mild")
    private String mild;

    @Column(name = "e_mild")
    private String eMild;

    @Column(name = "severe")
    private String severe;

    @Column(name = "e_severe")
    private String eSevere;

    @Column(name = "preventive")
    private String preventive;

    @Column(name = "e_preventive")
    private String ePreventive;

    @Column(name = "caution")
    private String caution;

    @Column(name = "e_caution")
    private String eCaution;

    @OneToMany(mappedBy = "diseaseConst", cascade = CascadeType.MERGE)
    private List<Drug> drugs;

    @OneToMany(mappedBy = "diseaseConst", cascade = CascadeType.MERGE)
    private List<Symptoms> symptoms;

    @OneToMany(mappedBy = "diseaseConst", cascade = CascadeType.MERGE)
    private List<Disease> diseases;
}

File: ai-server/app/models/disease_const.py
from sqlalchemy import BigInteger, Column, Integer, String
from sqlalchemy.orm import relationship

from app.db.base import Base


class DiseaseConst(Base):
    __tablename__ = "disease_const"

    id = Column(Integer, primary_key=True, autoincrement=True)

    rating = Column(BigInteger)
    e_rating = Column(BigInteger)

    name = Column(String)
    e_name = Column(String)

    description = Column(String)
    e_description = Column(String)

    type = Column(String)
    e_type = Column(String)

    site = Column(String)
    e_site = Column(String)

    reason = Column(String)
    e_reason = Column(String)

    mild = Column(String)
    e_mild = Column(String)

    severe = Column(String)
    e_severe = Column(String)

    preventive = Column(String)
    e_preventive = Column(String)

    caution = Column(String)
    e_caution = Column(String)

    drugs = relationship("Drug", back_populates="disease_const", cascade="merge")
    symptoms = relationship("Symptoms", back_populates="disease_const", cascade="merge")
    diseases = relationship("Disease", back_populates="disease_const", cascade="merge")
